lib/wegift/models/remote_code.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'uri'
+
+module Wegift
+  class RemoteCode < Response
+    attr_reader :url
+
+    # response/success
+    attr_reader :amount, :barcode_format, :barcode_string, :code, :expiry_date,
+      :pin, :type
+
+    def initialize(url:)
+      @url = url
+    end
+
+    def self.get(url)
+      uri = URI(url)
+      conn = Faraday.new(url: "#{uri.scheme}://#{uri.host}")
```

Comment:
From the Tests I could see it's on wegift.io - why not use the Wegift::Client here? You can pass @api_path as empty string the URL as `public/gifts/instant`

Reply:
we could but as the endpoint is public, there's no need to send credentials there and I decided not to

Reply:
also, we can't guarantee the URL returned will always be pointing to the same as we've set for the `Wegift::Client`

Reply:
we could sending the auth optional - I think that's less effort than rewriting the request - especially when we implement new non authed endpoints. 
And about the URL, is that somewhere documented?

Reply:
the URL is not documented, we only use what they send back to us when passing `url-instant` for the delivery format

Reply:
the only pieces of documentation we have is attached to the story card (8701)